// @dicebear/initials/lib/index.js
/*!
 * DiceBear Initials (@dicebear/initials)
 *
 * Code licensed under MIT (https://github.com/dicebear/dicebear/blob/v4/packages/initials/LICENSE)
 * Copyright (c) 2022 Florian Körner
 */
import { getInitials } from './utils/initials.js';
export const meta = {
    title: 'Initials',
    creator: 'Florian Körner',
    source: 'https://github.com/dicebear/dicebear',
    license: {
        name: 'CC0 1.0',
        url: 'https://creativecommons.org/publicdomain/zero/1.0/',
    },
};
export const create = ({ prng, options }) => {
    var _a, _b, _c, _d, _e;
    const fontFamily = (_b = (_a = options.fontFamily) === null || _a === void 0 ? void 0 : _a.join(', ')) !== null && _b !== void 0 ? _b : 'Arial, sans-serif';
    const fontSize = (_c = options.fontSize) !== null && _c !== void 0 ? _c : 50;
    const fontWeight = (_d = options.fontWeight) !== null && _d !== void 0 ? _d : 400;
    const seedInitials = getInitials(prng.seed.trim()).slice(0, (_e = options.chars) !== null && _e !== void 0 ? _e : 2);
    // prettier-ignore
    const svg = [
        `<text x="50%" y="50%" font-family="${fontFamily}" font-size="${fontSize}" fontWeight="${fontWeight}" fill="#FFF" text-anchor="middle" dy="${(fontSize * .356).toFixed(3)}">${seedInitials}</text>`,
    ].join('');
    return {
        attributes: {
            viewBox: '0 0 100 100',
        },
        body: svg,
    };
};
export { schema } from './schema.js';
